### PR TITLE
Add table azure_compute_ssh_key

### PR DESCRIPTION
**Pull request by srgg**
Adds support for SSH Publik keys, see https://learn.microsoft.com/en-us/cli/azure/sshkey?view=azure-cli-latest

# Example query results
<details>
  <summary>Results</summary>
  
```
select * from azure_compute_ssh_key;
[
 {
  "_ctx": {
   "connection_name": "azure"
  },
  "akas": [
   "azure:///subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/DEMO_GROUP/providers/Microsoft.Compute/sshPublicKeys/test-key.pub",
  ],
  "cloud_environment": "AzurePublicCloud",
  "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/DEMO_GROUP/providers/Microsoft.Compute/sshPublicKeys/test-key.pub",
  "name": "test-key.pub",
  "public_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC0XknryLEsmUxhl994+0bUsYJJ2DDTYlMIGNB1MecCAipmifIcF7GQ/heu1rvDjKt1EI8WNhzRbRHHdBWz4t3f3DW9EpVIk0cJKgzdl74T+2i8tP3FDQB5KyNWnqg23MrsY6tU3PubcNYOo1fxXjhWiP1BBavkNQ7kQJ3xBo0nl2xjYpowZ9SqXrwAP2I/0yFUt7JTPt1mDpXeBcYLu/f4wi/owiXdHOxmyS1/Mfs33mBQpFNS4lSMhhF+dO1Z6ToovD5xpDiAXQ/bLc+p4Nu3lRql1ycNHmE4B3zL7YV0v2m9SQDkMf3rrCh8yblMgDtTc8tN0AcOHqWQi36NLGutP9PJqKgqfBzuvNW+AyWgrYYcs0IrFTu2G66ep9KQeJm9d3zc3kV0I/0flxwBxq1H1m7lDLTJUvmvQGk1D20x6GmG1hGpNntwGxyKKnGWR0kvpf48BUnWyfS8gnxXI2UehmFyKAEda2ZUfSsagJMu1EX4u0uN5264DkMZ7pAfq0= davefitzjarrell@full-power-1.lan",
  "region": "eastus",
  "resource_group": "demo_group",
  "subscription_id": "00000000-0000-0000-0000-000000000000",
  "tags": {
   "SimpleTag": "test-upload"
  },
  "title": "test-key.pub",
  "type": null
 }
]
```
</details>

### CODE REVIEW

**Comment by rajlearner17** [comment]
@srgg I Appreciate using Steampipe and raising this PR!

We will review this and revert to you if we need further input.

File: docs/tables/table_azure_compute_ssh_key.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+### Retrieve SSH public key by name
+
+```sql
+select
+  name,
+  publicKey
+from
+    table_azure_compute_ssh_key
+where
+  name = 'key-name.';
```
[the remark this answers is not in the document]
Comment (misraved) [reply docs/tables/table_azure_compute_ssh_key.md:16]: @srgg Could you please add 2-3 more examples for this table that would highlight its usage?

**Review by misraved — COMMENTED** [review]
Thanks @srgg for adding this table. The structure looks good 👍.

I have left a few minor review comments, please let me know if you have any questions 👍 .

**Comment by srgg** [comment]
@rajlearner17 Yes, I do have plans. A bit busy. The plan is to update this PR on these weekends, thank you

File: docs/tables/table_azure_compute_ssh_key.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (srgg) [reply docs/tables/table_azure_compute_ssh_key.md:16]: @misraved Honestly, I have no idea how these ssh keys can be correlated with the rest of compute related data. So far, I only needed to gather them. Hopefully, in 2-3 weeks, I will know better

File: azure/table_azure_compute_ssh_key.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+				ShouldIgnoreErrorFunc: isNotFoundError([]string{"ResourceGroupNotFound", "ResourceNotFound", "404"}),
+			},
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listAzureComputeSshKeys,
+		},
+		Columns: azureColumns([]*plugin.Column{
+			{
+				Name:        "id",
+				Description: "The unique id identifying the resource in subscription",
```
[the remark this answers is not in the document]
Comment (srgg) [reply azure/table_azure_compute_ssh_key.go]: Done

File: azure/table_azure_compute_ssh_key.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromGo(),
+			},
+			{
+				Name:        "name",
+				Description: "Name of the ssh key",
```
[the remark this answers is not in the document]
Comment (srgg) [reply azure/table_azure_compute_ssh_key.go]: Done

**Comment by srgg** [comment]
@rajlearner17 @misraved and those who could be interested - PR is updated; the only issue I have with the last review comment. Thanks

File: docs/tables/table_azure_compute_ssh_key.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (khushboo9024) [reply docs/tables/table_azure_compute_ssh_key.md:16]: @srgg could you please add the following query to the table doc? 
```
### List compute virtual machines using SSH public key

```sql
select
  m.name as machine_name,
  k.name as ssh_key_name
from
  azure_compute_virtual_machine as m,
  jsonb_array_elements(linux_configuration_ssh_public_keys) as s
  left join azure_compute_ssh_key as k on k.public_key = s ->> 'keyData';
```
```

[the remark this answers is not in the document]
Comment (srgg) [reply docs/tables/table_azure_compute_ssh_key.md:16]: Done

File: azure/table_azure_compute_ssh_key.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+	}
+}
+
+//// LIST FUNCTION ////
+
+func listAzureComputeSshKeys(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listAzureComputeSshKeys")
+	session, err := GetNewSession(ctx, d, "MANAGEMENT")
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_compute_ssh_key. listAzureComputeSshKeys", "client_error", err)
```
Comment (misraved) [inline azure/table_azure_compute_ssh_key.go:94]:
```suggestion
		plugin.Logger(ctx).Error("azure_compute_ssh_key.listAzureComputeSshKeys", "client_error", err)
```

File: azure/table_azure_compute_ssh_key.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+		return nil, err
+	}
+
+	subscriptionID := session.SubscriptionID
+	client := compute.NewSSHPublicKeysClientWithBaseURI(session.ResourceManagerEndpoint, subscriptionID)
+	client.Authorizer = session.Authorizer
+	result, err := client.ListBySubscription(ctx)
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_compute_ssh_key. listAzureComputeSshKeys", "query_error", err)
```
Comment (misraved) [inline azure/table_azure_compute_ssh_key.go:103]:
```suggestion
		plugin.Logger(ctx).Error("azure_compute_ssh_key.listAzureComputeSshKeys", "query_error", err)
```

File: azure/table_azure_compute_ssh_key.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+func getAzureComputeSshKey(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAzureComputeSshKey")
+
+	name := d.KeyColumnQuals["name"].GetStringValue()
+	resourceGroup := d.KeyColumnQuals["resource_group"].GetStringValue()
+
+	session, err := GetNewSession(ctx, d, "MANAGEMENT")
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_compute_ssh_key. getAzureComputeSshKey", "client_error", err)
```
Comment (misraved) [inline azure/table_azure_compute_ssh_key.go:145]:
```suggestion
		plugin.Logger(ctx).Error("azure_compute_ssh_key.getAzureComputeSshKey", "client_error", err)
```

File: azure/table_azure_compute_ssh_key.go [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+		return nil, err
+	}
+	subscriptionID := session.SubscriptionID
+	client := compute.NewSSHPublicKeysClientWithBaseURI(session.ResourceManagerEndpoint, subscriptionID)
+	client.Authorizer = session.Authorizer
+
+	op, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_compute_ssh_key. getAzureComputeSshKey", "query_error", err)
```
Comment (misraved) [inline azure/table_azure_compute_ssh_key.go:154]:
```suggestion
		plugin.Logger(ctx).Error("azure_compute_ssh_key.getAzureComputeSshKey", "query_error", err)
```

File: azure/table_azure_compute_ssh_key.go [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+		Columns: azureColumns([]*plugin.Column{
+			{
+				Name:        "id",
+				Description: "The unique ID identifying the resource in subscription.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromGo(),
+			},
+			{
+				Name:        "name",
+				Description: "Name of the ssh key.",
```
Comment (misraved) [inline azure/table_azure_compute_ssh_key.go:37]:
```suggestion
				Description: "Name of the SSH key.",
```

**Review by misraved — COMMENTED** [review]
My apologies @srgg for adding incorrect suggestions in the initial round of review. Please make the minor updates to the log messages and then we can go ahead with the release 👍.

**Comment by srgg** [comment]
> My apologies @srgg for adding incorrect suggestions in the initial round of review. Please make the minor updates to the log messages and then we can go ahead with the release 👍.

@misraved No worries, I've introduced the requested changes. Thank you for the review

**Review by misraved — APPROVED** [review]

**Pull request merged**